frontend/src/metabase/query_builder/actions.js

```diff
@@ -1383,6 +1392,27 @@ export const queryCompleted = (question, queryResults) => {
     }
 
     dispatch.action(QUERY_COMPLETED, { card, queryResults });
+
+    dispatch.action(SET_SHOW_LOADING_COMPLETE_FAVICON, true);
```

Comment:
Ok so there are 3 pieces here. Could we extract them to 3 separate thunks to keep this action easier to read? e.g.

```
// at the start
dispatch(enableLoadingControls());

// when the query completes
dispatch(disableLoadingControls());

// at the end
dispatch(showLoadingControls());
```

Naming can be arbitary
```

Reply:
And I really like this implementation, could we update dashboards to something similar? Thanks!

Reply:
@ranquild We can certainly adjust the Dashboard stuff to be more similar. I'm refactoring it now, it's a little different because of the x/y loaded, so keeping the title in state isn't as straight forward. However, it would be easy to move that logic into a selector and keep the `DashboardApp` component nice and tidy.

Reply:
Scratch that. Dashboard actions now update state in redux with the dashboard titles during the loading process.